oeps/processes/oep-0055/decisions/0001-use-backstage-to-support-maintainers.rst

```diff
@@ -0,0 +1,133 @@
+0001 Use Backstage to Support Maintainers
+#########################################
+
+Status
+******
+
+**Draft**
+
+.. TODO Upon Acceptance
+
+	* Update OEP to mention the catalog-info.yaml file instead of the readme.
+		* Relevant link https://open-edx-proposals.readthedocs.io/en/latest/processes/oep-0055-proc-project-maintainers.html#documenting-maintainers
+
+Context
+*******
+
+* :doc:`OEP-55 <../../oep-0055-proc-project-maintainers>` laid out that we should document the maintainers of a component.
+
+* As a maintainer it is hard to know what components you are maintaining.
+
+* As a community member it is currently difficult to quickly look up who the
+  maintainers are for multiple components quickly.
+
+* Given the number of repos that make up the Open edX platform and the
+  complexity of the job of maintaining them, we need a system for cataloging,
+  and making transparent the critical details of our repositories.
+
+Decision
+********
+
+To solve the above issues and to enable the potential to ease other
+discoverability and maintainership burdens, the Open edX community shall
+maintain a `backstage`_ instance and related metadata.
+
+Consequences
+************
+
+* To support the backstage instance ownership data will have to be stored in a
+  structured way.  We will follow Backstage best practices and store this data
+  in a ``catalog-info.yaml`` file in each component repository.
+
+* Once a backstage instance exists we have the opportunity to further leverage
+  it as a centralized access point for maintainers as well as community members
+  to quickly and easily assess different components and find relevant links to
+  code, documentation, owners, etc.
+
+* Backstage will also give us the ability to expose other health metrics to
+  maintainers so they can more easily assess the components they own.
+
+* As a result of adding the ``catalog-info.yaml`` file we end with two top-level
+  files that have very similar data.  If this decision is not reversed after the
+  initial pilot, it may make sense to revisit this consequence and see if we can
+  consolidate this metadata into a single file.
+
+Rejected Alternatives
+*********************
+
+Spreadsheet
+===========
+
+Pros
+----
+
+* Easy to build and get a high level view of the system.
+
+Cons
+----
+
+* Not a beautiful UI for maintainers to work with.
+
+* Data overload as we put more and more data into it with little to no control
+  over the presentation of the data.
+
+Owners in the Readme
+====================
+
+Pros
+----
+
+* Easy to see while in the repo.
+
+Cons
+----
+
+* Harder to aggregate the data from the unstructured files in a way that would
+  allow us to easily build a `single pane of glass`_ where maintainers can see
+  all the things they own.
```

Comment:
For some reason I have in my memory something about using github repo tags for this? I think it was @ormsbee's idea. In any case, would we want to integrate that with Backstage somehow or are they just separate concerns?

Reply:
Your memory serves you right @hurtstotouchfire -- on the OEP-57 draft, [Dave mentioned using GH topics](https://github.com/openedx/open-edx-proposals/pull/312#discussion_r819157290) to group repositories into tiers like Core, Supporting, etc. Repo topics are an interesting way of storing metadata in a GH-native way that you might consider employing as you iterate on this @feanil .

Correct me if I'm wrong, but I get the sense that you are hoping to get this ADR over the line so that we can experiment with actually using backstage for the maintainer pilot. After we use backstage for a while it'll probably be easier to tell which GH features we do and don't want to tie in with, as well as what exactly the schema should look like. That is to say: I'm fine if you're not explicitly rejecting any alternatives at the moment, since we're not exactly sure what this thing will look like and how everyone will use it.